src/website/views/DashboardContainer.tsx

```diff
@@ -101,48 +114,77 @@ class DashboardContainer extends React.PureComponent<DashboardContainerProps> {
   }
 
   _haversineDistance(lat1: number, long1: number, lat2: number, long2: number) {
-
-    function toRadians(angle: number): number{
-      return angle * Math.PI / 180
+    function toRadians(angle: number): number {
+      return (angle * Math.PI) / 180;
     }
 
-    const EARTH_RADIUS = 6571 // in km
+    const EARTH_RADIUS = 6571; // in km
 
-    let delta_lat = lat2-lat1
-    let delta_lat_rad = toRadians(delta_lat)
-    let delta_long = long2-long1
-    let delta_long_rad = toRadians(delta_long)
+    let delta_lat = lat2 - lat1;
+    let delta_lat_rad = toRadians(delta_lat);
+    let delta_long = long2 - long1;
+    let delta_long_rad = toRadians(delta_long);
 
-    let a = (Math.sin(delta_lat_rad/2) * Math.sin(delta_lat_rad/2))
-    + (Math.cos(toRadians(lat1)) * Math.cos(toRadians(lat2))
-    * Math.sin(delta_long_rad/2)
-    * Math.sin(delta_long_rad/2))
-    let c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1-a));
+    let a =
+      Math.sin(delta_lat_rad / 2) * Math.sin(delta_lat_rad / 2) +
+      Math.cos(toRadians(lat1)) *
+        Math.cos(toRadians(lat2)) *
+        Math.sin(delta_long_rad / 2) *
+        Math.sin(delta_long_rad / 2);
+    let c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
 
     let distance = EARTH_RADIUS * c;
 
-    return distance
+    return distance;
   }
 
   _computeTotalTripDistance(latitude: number[], longitude: number[]) {
-    if(latitude.length != longitude.length){
+    if (latitude.length != longitude.length) {
       return -1;
     }
 
     let totalDistance = 0;
 
-    for(let i = 1; i < latitude.length; i++){
-        totalDistance += this._haversineDistance(latitude[i-1], longitude[i-1], latitude[i], longitude[i]);
+    for (let i = 1; i < latitude.length; i++) {
+      totalDistance += this._haversineDistance(
+        latitude[i - 1],
+        longitude[i - 1],
+        latitude[i],
+        longitude[i],
+      );
     }
 
     return Number(totalDistance.toFixed(2));
   }
+
+  _validTimestamp(timestampISO: number) {
+    let cond1 =
+      timestampISO >=
+      this._parseISOString(this.props.timestamp.timestamp.startDate);
+    let cond2 =
+      timestampISO <=
+      this._parseISOString(this.props.timestamp.timestamp.endDate);
+
+    if (
+      this.props.timestamp.timestamp.startDate == null &&
+      this.props.timestamp.timestamp.endDate == null
+    ) {
+      return true;
+    }
```

Comment:
Let's have this if condition at the top of the function so we don't need to declare variables in case both are `null`. Also, what happens here when one of `startDate` and `endDate` is not null? Will it still work?

Reply:
Gotcha